identifier/identifier.go

```diff
@@ -56,3 +105,80 @@ func NewIP(ip netip.Addr) ACMEIdentifier {
 		Value: ip.StringExpanded(),
 	}
 }
+
+// FromCert extracts the Subject Common Name and Subject Alternative Names from
+// a certificate, and returns a slice of ACMEIdentifiers and an error.
+//
+// FromCSR is nearly identical but handles CSRs, and is kept separate so that
+// it's always clear we are handling an untrusted CSR.
+func FromCert(cert *x509.Certificate) ([]ACMEIdentifier, error) {
+	var sans []ACMEIdentifier
+	for _, name := range cert.DNSNames {
+		sans = append(sans, NewDNS(name))
+	}
+	if cert.Subject.CommonName != "" {
+		// Boulder won't generate certificates with a CN that's not also present
+		// in the SANs, but such a certificate is possible. If appended, this is
+		// deduplicated later with Normalize(). We assume the CN is a DNSName,
+		// because CNs are untyped strings without metadata, and we will never
+		// configure a Boulder profile to issue a certificate that contains both
+		// an IP address identifier and a CN.
+		sans = append(sans, NewDNS(cert.Subject.CommonName))
+	}
+
+	for _, netIP := range cert.IPAddresses {
+		netipAddr, ok := netip.AddrFromSlice(netIP)
+		if !ok {
+			return nil, fmt.Errorf("converting IP from bytes: %s", netIP)
+		}
+		sans = append(sans, NewIP(netipAddr))
+	}
```

Comment:
Here we're parsing the IP address, then `NewIP` is un-parsing it. So the effect is just that we're validating it. We don't validate DNSNames in this function; that's the role of other functions. I recommend:

```suggestion
	for _, ip := range cert.IPAddresses {
		sans = append(sans, ACMEIdentifier{
		  Type: TypeIP,
		  Value: ip.String(),
		})
	}
```

That also allows us to remove the error return of `FromCert()`, which is nice.